packages/plugins/i18n/admin/src/components/BulkLocaleActionModal.tsx

```diff
@@ -0,0 +1,258 @@
+      .map(([key, value]) => {
+        if (isMessageDescriptor(value?.defaultMessage)) {
+          return `${key}: ${formatMessage(value.defaultMessage)}`;
+        } else if (isMessageDescriptor(value)) {
+          return `${key}: ${formatMessage(value)}`;
+        }
+
+        return `${key}: ${value}`;
+      })
+      .join(' ');
+
+    return (
+      <Flex gap={2}>
+        <CrossCircle fill="danger600" />
+        <Tooltip description={validationErrorsMessages}>
+          <TypographyMaxWidth textColor="danger600" variant="omega" fontWeight="semiBold" ellipsis>
+            {validationErrorsMessages}
+          </TypographyMaxWidth>
+        </Tooltip>
+      </Flex>
+    );
+  }
+
+  if (status === 'published') {
+    return (
+      <Flex gap={2}>
+        <CheckCircle fill="success600" />
+        <Typography textColor="success600" fontWeight="bold">
+          {formatMessage({
+            id: 'content-manager.bulk-publish.already-published',
+            defaultMessage: 'Already Published',
+          })}
+        </Typography>
+      </Flex>
+    );
+  }
+
+  if (status === 'modified') {
+    return (
+      <Flex gap={2}>
+        <ArrowsCounterClockwise fill="alternative600" />
+        <Typography>
+          {formatMessage({
+            id: 'app.utils.ready-to-publish-changes',
+            defaultMessage: 'Ready to publish changes',
+          })}
+        </Typography>
+      </Flex>
+    );
+  }
+
+  return (
+    <Flex gap={2}>
+      <CheckCircle fill="success600" />
+      <Typography>
+        {formatMessage({
+          id: 'app.utils.ready-to-publish',
+          defaultMessage: 'Ready to publish',
+        })}
+      </Typography>
+    </Flex>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * BoldChunk
+ * -----------------------------------------------------------------------------------------------*/
+
+const BoldChunk = (chunks: React.ReactNode) => <Typography fontWeight="bold">{chunks}</Typography>;
+
+/* -------------------------------------------------------------------------------------------------
+ * BulkLocaleActionModal
+ * -----------------------------------------------------------------------------------------------*/
+
+interface BulkLocaleActionModalProps {
+  rows: LocaleStatus[];
+  headers: {
+    label: string;
+    name: string;
+  }[];
+  onClose: () => void;
+  localesMetadata: Locale[];
+  validationErrors?: Record<
+    Modules.Documents.Params.Locale.StringNotation,
+    Record<string, MessageDescriptor>
+  >;
+}
+
+const BulkLocaleActionModal = ({
+  headers,
+  rows,
+  localesMetadata,
+  onClose,
+  validationErrors = {},
+}: BulkLocaleActionModalProps) => {
+  const { formatMessage } = useIntl();
+  const navigate = useNavigate();
+
+  const selectedRows: LocaleStatus[] = useTable(
+    'BulkLocaleActionModal',
+    (state: { selectedRows: LocaleStatus[] }) => state.selectedRows
+  );
```

Comment:
```suggestion
  const selectedRows: LocaleStatus[] = useTable<LocaleStatus[]>(
    'BulkLocaleActionModal',
    (state) => state.selectedRows
  );
```

I _think_ this should work...